frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -519,12 +519,19 @@ pub mod pallet {
 		/// Maximum number of iteration of balancing that will be executed in the embedded miner of
 		/// the pallet.
 		type MinerMaxIterations: Get<u32>;
+
 		/// Maximum weight that the miner should consume.
 		///
 		/// The miner will ensure that the total weight of the unsigned solution will not exceed
-		/// this values, based on [`WeightInfo::submit_unsigned`].
+		/// this value, based on [`WeightInfo::submit_unsigned`].
 		type MinerMaxWeight: Get<Weight>;
 
+		/// Maximum length (bytes) that the mined solution should consume.
+		///
+		/// The miner will ensure that the total length of the unsigned solution will not exceed
+		/// this value.
+		type MinerMaxLength: Get<u32>;
```

Comment:
How about we merge these into `type MinerLimits: Get<(u32, u32)>`?

Reply:
This turns out to make things pretty complicated for tests which want to mutate these values independently. For now I'm leaning toward leaving them independent.

Reply:
What is the complication for test? I think might be due to a misunderstanding.

Reply:
```
frame_support::parameter_types! {
    pub static MinerLimits: (Weight, u32) = (x, y);
}

impl Builder {
  fn limits(mut self, weight: Weight, len: u32) -> Self {
    MinerLimits::set((weight, len));
    self
  }
  
  // and maybe two functions that set len and weight independently, preserving the other one. 
  ```
  
  Where's the catch? 🤨

Reply:
I got that far, but then we have things like this:

https://github.com/paritytech/substrate/blob/78ddb641fdf6452b817d82778d45c377cd7d3008/frame/election-provider-multi-phase/src/unsigned.rs#L761-L762

We can't update the weight as part of the builder constructor, which means we have to update it in the runtime. I'm certain that it's possible to mutate part of a compound storage value efficiently. However, figuring out precisely how to do it was turning into a time sink, and this didn't feel like a huge deal, which is when I wrote the note above.

Reply:
> I'm certain that it's possible to mutate part of a compound storage value efficiently

Indeed, and note that this is testing, no point in caring for efficiency. 

(Also this is not a storage item anyway, so probabyl you are confusing it with something else). 

> However, figuring out precisely how to do it was turning into a time sink

I'm glad you took this option, don't want you to waste too much time, but now that I am here: 

ugly, but what about this if you want to update only the first element for example: `<MinerLimits>::set(<MinerLimits>::get().0, new_val)`? and then you can have 3 functions in total, as I recommended above: 

```
impl Builder {
  fn limits(mut self, weight: Weight, len: u32) -> Self {
    MinerLimits::set((weight, len));
    self
  }
  fn weight_limit(mut self, weight: Weight) -> Self {
    MinerLimits::set((weight, MinerLimits::get().1));
  }
  fn len_limit(mut self, len: u32) -> Self {
    MinerLimits::set((MinerLimits::get().0, len));
  }


  
  // and maybe two functions that set len and weight independently, preserving the other one. 
  ```

Reply:
@coriolinus how about this? If you want, I can push and try and fit myself. Maybe I am missing something.

Reply:
Sure, thanks. TBH I haven't worked on this bit in particular for a while; I'd interpreted your previous comment as acknowledgement that it wasn't worth combining the two. If it turns out to be straightforward, I'd be interested to see how it's accomplished.

Reply:
Let's deal with this later.